soc: xtensa: keep the CAVS old flasher working

enjiamai opened the pull request:
Keep the cavstwist.sh it is still working by specifying an
environment variable when CAVS_OLD_FLASHER is set. Provide
a fallback once the client-server-based tool does not work
well in some conditions.

Signed-off-by: Enjia Mai <enjia.mai@intel.com>

laurenmurphyx64 commented:
Set up my old environment (+ `CAVS_LEGACY_MODE=y`) and tested this out with cavstwist.sh + `samples/hello_world` and Twister + `tests/kernel/fatal/common` and it worked a-OK. :smile:

laurenmurphyx64 commented on boards/xtensa/intel_adsp_cavs15/board.cmake:
```diff
@@ -1,5 +1,11 @@
 # SPDX-License-Identifier: Apache-2.0
 
+set(LEAGCY_MODE $ENV{CAVS_LEGACY_MODE})
```
LEAGACY -> LEGACY

enjiamai replied:
Sorry, thanks so much for correcting it.

enjiamai replied:
correct it, thanks!

laurenmurphyx64 reviewed (commented):
I also think we should make a note somewhere in the public documentation on how to turn on legacy mode. Aside from those two comments, this looks good to go.

enjiamai commented:
> I also think we should make a note somewhere in the public documentation on how to turn on legacy mode. Aside from those two comments, this looks good to go.

Yes, I think this is necessary. Will add them ASAP.

andyross reviewed (commented):
Maybe call this something more descriptive than "legacy"?  Even something like "OLD_FLASHER" would provide better detail.

enjiamai commented:
> Maybe call this something more descriptive than "legacy"? Even something like "OLD_FLASHER" would provide better detail.

Thanks for the review! Let's call it old flasher.

edersondisouza reviewed (commented):
Maybe add this info to cavstwist.sh? I'm imagining people will try to use cavstwist.sh when the new cavstool.py fails, but then cavstwist.sh will fail and there's no info about this new env var anywhere.

enjiamai commented:
> Maybe add this info to cavstwist.sh? I'm imagining people will try to use cavstwist.sh when the new cavstool.py fails, but then cavstwist.sh will fail and there's no info about this new env var anywhere.

Sorry, @edersondisouza I missing this msg,  I will add it right away, thanks!

teburd reviewed (dismissed):
Thanks for keeping this working, cavstool is half the test case in many scenarios, and while developing test cases cavstwist is quite nice

enjiamai committed:
soc: xtensa: keep the CAVS old flasher still works

Keep the cavstwist.sh it is still working by specifying an
environment variable when CAVS_OLD_FLASHER is set. Provide
a fallback once the client-server-based tool does not work
well in some conditions.

Signed-off-by: Enjia Mai <enjia.mai@intel.com>

enjiamai commented on boards/xtensa/intel_adsp_cavs15/board.cmake:
```diff
@@ -1,5 +1,11 @@
 # SPDX-License-Identifier: Apache-2.0
 
+set(OLD_FLASHER $ENV{CAVS_OLD_FLASHER})
+if(${OLD_FLASHER})
```
Yes, the will be simplified, thanks for pointing this out. Updated, thanks!
